Assets/SteamVR_Unity_Toolkit/Scripts/Controls/2D/RadialMenu/RadialMenu.cs

```diff
@@ -0,0 +1,300 @@
+    public List<RadialMenuButton> Buttons;
+	public GameObject buttonPrefab;
+	[Range(0f,1f)]
+	public float buttonThickness = 0.5f;
+	public Color buttonColor = Color.white;
+	public float offsetDistance = 1;
+	[Range(0,359)]
+	public float offsetRotation;
+
+	public bool rotateIcons;
+	public float iconMargin;
+    public bool isShown;
+    public bool HideOnRelease;
+
+    //Has to be public to keep state from editor -> play mode?
+	public List<GameObject> menuButtons;
+
+    float currentTouchAngle;
+    int currentHover = -1;
+    int currentPress = -1;
+    #endregion
+
+    #region Unity Methods
+
+    void Start()
+    {
+        if (Application.isPlaying)
+        {
+            if(!isShown)
+                transform.localScale = Vector3.zero;
+            RegenerateButtons();
+        }
+            
+    }
+
+    void Update()
+    {
+        //Keep track of pressed button and constantly invoke Hold event
+        if (currentPress != -1) 
+        {
+            Buttons[currentPress].Press();
+        }
+    }
+
+    #endregion
+
+    #region Interaction
+
+    //Turns and Angle and Event type into a button action
+    void InteractButton(float angle, ButtonEvent evt) //Can't pass ExecuteEvents as parameter? Unity gives error
+    {
+        //Get button ID from angle
+        float buttonAngle = 360f / Buttons.Count; //Each button is an arc with this angle
+        angle = mod((angle + offsetRotation), 360); //Offset the touch coordinate with our offset
+        currentTouchAngle = angle; //Keep track of our angle for outside use if we want it later
+        int buttonID = (int)mod(((angle + (buttonAngle / 2f)) / buttonAngle), Buttons.Count); //Convert angle into ButtonID (This is the magic)
+        var pointer = new PointerEventData(EventSystem.current); //Create a new EventSystem (UI) Event
+
+        //If we changed buttons while moving, un-hover and un-click the last button we were on
+        if (currentHover != buttonID && currentHover != -1)
+        {
+            ExecuteEvents.Execute(menuButtons[currentHover], pointer, ExecuteEvents.pointerUpHandler);
+            ExecuteEvents.Execute(menuButtons[currentHover], pointer, ExecuteEvents.pointerExitHandler);
+        }
+        if (evt == ButtonEvent.click) //Click button if click, and keep track of current press
+        {
+            ExecuteEvents.Execute(menuButtons[buttonID], pointer, ExecuteEvents.pointerDownHandler);
+            currentPress = 1;
+            Buttons[buttonID].Click();
+        }
+        else if (evt == ButtonEvent.unclick) //Clear press id to stop invoking OnHold method
+        {
+            ExecuteEvents.Execute(menuButtons[buttonID], pointer, ExecuteEvents.pointerUpHandler);
+            currentPress = -1;
+        }
+        else if (evt == ButtonEvent.hoverOn && currentHover != buttonID) // Show hover UI event (darken button etc)
+            ExecuteEvents.Execute(menuButtons[buttonID], pointer, ExecuteEvents.pointerEnterHandler);
```

Comment:
for consistency it may be worth putting brackets around the else if block as well

all other places that have a single line of code in the conditional use braces just for neatness.